src/main/java/dev/jbang/cli/Edit.java

```diff
@@ -220,17 +209,19 @@ File createProjectForEdit(Script script, boolean reload) throws IOException {
 		Path srcFile = srcDir.toPath().resolve(name);
 		Util.createLink(srcFile, originalFile.toPath());
 
-		for (Source source : script.getResolvedSources()) {
-			File sfile = null;
-			if (source.getJavaPackage().isPresent()) {
-				File packageDir = new File(srcDir, source.getJavaPackage().get().replace(".", File.separator));
-				packageDir.mkdir();
-				sfile = new File(packageDir, source.getResolvedPath().toFile().getName());
-			} else {
-				sfile = new File(srcDir, source.getResolvedPath().toFile().getName());
+		if (script.runUnit instanceof Script) {
```

Comment:
when is script never a Script ?

Reply:
When it's a JAR. In fact, in the current version of Jbang you can do `jbang edit /path/to/a/jar.jar` and it will accept it and then fail with an ugly error. (This doesn't fix that, but it shows you can get to that code and not have an instance of Script)